vispy/visuals/bar.py

```diff
@@ -26,23 +26,39 @@ class BarVisual(MeshVisual):
     color : instance of Color
         Color of the bar.
     orientation : {'h', 'v'}
-        Orientation of the histogram - 'h' is default
+        Orientation of the bars - 'h' is default
+    color_array : array-like
+        [[1, 0, 0], [0, 1, 0], [0, 0, 1]] exactly one rgb array for each bar. This would be for 3 Bars 
     """
 
-    def __init__(self, height, bottom=None, width=0.8, shift=0.0, color='w', orientation='h'):
+    def __init__(self, height, bottom=None, width=0.8, shift=0.0, color='w', orientation='h', color_array=None):
         if bottom is None:
             bottom = np.zeros(height.shape[0])
+
+        if height.shape != bottom.shape:
+            raise ValueError("Height and Bottom must be same shape: Height: " + height.shape + " Bottom: " + bottom.shape)
         
         rr, tris = calc_vertices(height, bottom, width, shift, orientation)
 
-        MeshVisual.__init__(self, rr, tris, color=color)
+        if color_array is not None and color_array.shape[0] == height.shape[0] and color_array.shape[1] == 3:
+            MeshVisual.__init__(self, rr, tris, face_colors=np.repeat(color_array, 2, axis=0))
+        else:
+            MeshVisual.__init__(self, rr, tris, color=color)
```

Comment:
Great! To get rid of this if clause (and get extra perks like string inputs and aplha values) I would use the `ColorArray` just like you did in the `Candle` PR.

Reply:
@Bliss3d you think you can get this to work with a single `color` argument? I think it would be more intuitive, kinda like `Markers` does in these few lines:

https://github.com/vispy/vispy/blob/f13d7da790030ef86f2969d04f9462d7d1d35e11/vispy/visuals/markers.py#L607-L613

As you can see, you can use the `ColorArray` object from `vispy.color`, which automatically takes care of all the weird cases (a string like `'w'`, a `(n, 3)` rgb array, or an `rgba`, and so on). You can then simply check if it's a single color, and if so pass it to `color`, otherwise pass it to `face_colors`.

Reply:
I'm guessing something like this